apsuite/optics_analysis/orbit_correction.py

```diff
@@ -0,0 +1,178 @@
+
+    def calc_jacobian_matrix(self, model=None):
+        """."""
+        if model is None:
+            model = self.model
+        self.respm.model = model
+        return self.respm.get_respm()
+
+    def _get_orbit_residue(self, model):
+        if self.dim == '4d':
+            cod = pyaccel.tracking.find_orbit4(model, indices='open')
+        elif self.dim == '6d':
+            cod = pyaccel.tracking.find_orbit6(model, indices='open')
+        codx = cod[0, self.bpm_idx].flatten()
+        cody = cod[2, self.bpm_idx].flatten()
+        res = np.hstack((codx, cody))
+        return res
+
+    def get_kicks(self, model=None, chidx=None, cvidx=None):
+        """Return corrector kicks."""
+        if model is None:
+            model = self.model
+        if chidx is None:
+            chidx = self.ch_idx
+        if cvidx is None:
+            cvidx = self.cv_idx
+        hkick = []
+        for mag in chidx:
+            hkick_seg = []
+            for seg in mag:
+                hkick_seg.append(model[seg].hkick_polynom)
+            hkick.append(sum(hkick_seg))
+        vkick = []
+        for mag in cvidx:
+            vkick_seg = []
+            for seg in mag:
+                vkick_seg.append(model[seg].vkick_polynom)
+            vkick.append(sum(vkick_seg))
+        hkick = np.array(hkick)
+        vkick = np.array(vkick)
+        return np.hstack((hkick, vkick))
+
+    def _set_delta_kicks(self, model=None, corridx=None, delta_kicks=None):
+        if model is None:
+            model = self.model
+        if corridx is None:
+            corridx = self.corr_idx
+        if delta_kicks is None:
+            raise Exception('Missing Delta Kicks values')
+        for idx_mag, mag in enumerate(corridx):
+            delta = delta_kicks[idx_mag]/len(mag)
+            for _, seg in enumerate(mag):
+                if idx_mag < self._nch:
+                    kick = model[seg].hkick_polynom + delta
+                    model[seg].hkick_polynom = kick
+                else:
+                    kick = model[seg].vkick_polynom + delta
+                    model[seg].vkick_polynom = kick
+
+    def _check_kicks(self, model=None, chidx=None, cvidx=None):
+        if model is None:
+            model = self.model
+        if chidx is None:
+            chidx = self.ch_idx
+        if cvidx is None:
+            cvidx = self.cv_idx
+
+        kicks = self.get_kicks(model, chidx, cvidx)
+        kicksx, kicksy = kicks[:self._nch], kicks[self._nch:]
+        kicksx_above = kicksx > OrbitCorr.MAX_HKICK
+        kicksy_above = kicksy > OrbitCorr.MAX_VKICK
+
+        if sum(kicksx_above) or sum(kicksy_above):
+            str_above = '\n'
+            str_above += 'HKick > MaxHKick at CHs: {0:s} \n'
+            str_above += 'VKick > MaxVKick at CVs: {1:s}'
+            xlist = str(np.argwhere(kicksx_above).flatten())
+            ylist = str(np.argwhere(kicksy_above).flatten())
+            raise Exception(str_above.format(xlist, ylist))
```

Comment:
I think it is better to raise a `ValueError`. `Exception` is too general.

Reply:
ok, changed.